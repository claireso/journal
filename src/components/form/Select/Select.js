import React from 'react'
import PropTypes from 'prop-types'

import { IconAngleRight } from '@components/Icons'

import { GroupInline } from '../Group'
import Label from '../Label'

import * as S from './Select.styles'

const Select = (props = {}) => {
  return (
    <GroupInline>
      <Label htmlFor={props.name}>{props.label}</Label>
      <S.SelectWrapper>
        <S.StyledSelect id={props.name} name={props.name} defaultValue={props.value} required>
          {props.options.map((option, index) => {
            return (
              <option key={index} value={option.value}>
                {option.label}
              </option>
            )
          })}
        </S.StyledSelect>
        <IconAngleRight />
      </S.SelectWrapper>
    </GroupInline>
  )
}

Select.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired
    })
  ).isRequired,
  value: PropTypes.string
}

export default Select
