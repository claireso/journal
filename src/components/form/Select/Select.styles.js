import styled from 'styled-components'

export const SelectWrapper = styled.div`
  position: relative;
  svg {
    position: absolute;
    top: 50%;
    right: 0.9rem;
    bottom: 0;
    transform: translate(0, -50%) rotate(90deg);
    pointer-events: none;
  }
`

export const StyledSelect = styled.select`
  appearance: none;
  background: var(--form-input-bg);
  border: 1px solid var(--border-color);
  border-radius: 0.5rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  cursor: pointer;
  font-family: 'Roboto', Arial, sans-serif;
  font-size: var(--font-size-normal);
  outline: none;
  padding: 1.1rem 2.6rem 1.1rem 1.1rem;
  &:focus {
    border-color: var(--border-color-focus);
  }
`
