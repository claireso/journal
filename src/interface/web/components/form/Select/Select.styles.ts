import { styled } from '@web/theme'

export const SelectWrapper = styled('div', {
  position: 'relative',
  svg: {
    position: 'absolute',
    top: '50%',
    right: '0.9rem',
    bottom: 0,
    transform: 'translate(0, -50%) rotate(90deg)',
    pointerEvents: 'none'
  }
})

export const StyledSelect = styled('select', {
  appearance: 'none',
  background: '$white',
  border: '1px solid $gray200',
  borderRadius: '0.5rem',
  boxShadow: '$2',
  cursor: 'pointer',
  fontFamily: '$sansSerif',
  fontSize: '$4',
  outline: 'none',
  padding: '1.1rem 2.6rem 1.1rem 1.1rem',
  transition: '$borderColor',
  '&:focus': {
    borderColor: '$gray300'
  }
})
