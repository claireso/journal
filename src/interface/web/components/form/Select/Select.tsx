import React from 'react'

import Icon from '@web/components/Icons'

import Group from '../Group'
import Label from '../Label'

import * as cls from './styles.css'

interface Option {
  label: string
  value: string
}

interface SelectProps {
  name: string
  label: string
  options: Option[]
  value?: string
}

const Select = ({ name, label, value, options }: SelectProps) => {
  return (
    <Group inline>
      <Label htmlFor={name}>{label}</Label>
      <div className={cls.wrapper}>
        <select className={cls.select} id={name} name={name} defaultValue={value} required>
          {options.map((option, index) => {
            return (
              <option key={index} value={option.value}>
                {option.label}
              </option>
            )
          })}
        </select>
        <div className={cls.icon}>
          <Icon name="arrow-down" size="sm" />
        </div>
      </div>
    </Group>
  )
}

export default Select
