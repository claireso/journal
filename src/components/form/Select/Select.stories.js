import React from 'react'

import Select from './index'

export default {
  title: 'Form/Select'
}

const options = [
  {
    label: 'Value 1',
    value: '1'
  },
  {
    label: 'Value 2',
    value: '2'
  }
]

export const Basic = () => <Select label="Label" name="position" options={options} />
