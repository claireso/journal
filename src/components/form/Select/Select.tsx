import React from 'react'

import { IconAngleRight } from '@components/Icons'

import { GroupInline } from '../Group'
import Label from '../Label'

import * as S from './Select.styles'

interface Option {
  label: string
  value: string
}

interface SelectProps {
  name: string
  label: string
  options: Option[]
  value?: string
}

const Select = ({ name, label, value, options }: SelectProps) => {
  return (
    <GroupInline>
      <Label htmlFor={name}>{label}</Label>
      <S.SelectWrapper>
        <S.StyledSelect id={name} name={name} defaultValue={value} required>
          {options.map((option, index) => {
            return (
              <option key={index} value={option.value}>
                {option.label}
              </option>
            )
          })}
        </S.StyledSelect>
        <IconAngleRight />
      </S.SelectWrapper>
    </GroupInline>
  )
}

export default Select
