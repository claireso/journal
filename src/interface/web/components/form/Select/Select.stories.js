import React from 'react'

import Select from './index'

const params = {
  title: 'Form/Select'
}

export default params

const options = [
  {
    label: 'Value 1',
    value: '1'
  },
  {
    label: 'Value 2',
    value: '2'
  }
]

export const Basic = () => <Select label="Label" name="position" options={options} />
