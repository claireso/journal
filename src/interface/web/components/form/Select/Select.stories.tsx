import type { Meta, StoryObj } from '@storybook/react-webpack5'

import Select from './Select'

const meta: Meta<typeof Select> = {
  title: 'Form/Select',
  component: Select
}

export default meta
type Story = StoryObj<typeof Select>

export const Basic: Story = {
  args: {
    label: 'Label',
    name: 'position',
    options: [
      {
        label: 'Value 1',
        value: '1'
      },
      {
        label: 'Value 2',
        value: '2'
      }
    ]
  }
}
