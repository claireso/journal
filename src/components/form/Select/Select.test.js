import { render } from '@testing-library/react'

import Select from './index'

describe('<Select />', () => {
  test('should render select', () => {
    const props = {
      label: 'My select',
      name: 'myselect',
      options: [
        {
          label: 'option 1',
          value: 'option1'
        },
        {
          label: 'option 2',
          value: 'option2'
        }
      ],
      value: 'option2'
    }

    const { container } = render(<Select {...props} />)

    expect(container).toMatchSnapshot()
  })
})
