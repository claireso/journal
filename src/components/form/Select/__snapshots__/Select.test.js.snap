// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render select 1`] = `
.c0 {
  margin: 0 0 var(--gutter);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > label {
  margin: 0 1rem 0 0;
}

.c2 {
  display: block;
  font-size: 1.6rem;
  line-height: 1.5;
  margin: 0 0 1.2rem;
}

.c3 {
  position: relative;
}

.c3 svg {
  position: absolute;
  top: 50%;
  right: 0.9rem;
  bottom: 0;
  -webkit-transform: translate(0,-50%) rotate(90deg);
  -ms-transform: translate(0,-50%) rotate(90deg);
  transform: translate(0,-50%) rotate(90deg);
  pointer-events: none;
}

.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: var(--form-input-bg);
  border: 1px solid var(--border-color);
  border-radius: 0.5rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  cursor: pointer;
  font-family: 'Roboto',Arial,sans-serif;
  font-size: var(--font-size-normal);
  outline: none;
  padding: 1.1rem 2.6rem 1.1rem 1.1rem;
}

.c4:focus {
  border-color: var(--border-color-focus);
}

<div>
  <div
    class="c0 c1"
  >
    <label
      class="c2"
      for="myselect"
    >
      My select
    </label>
    <div
      class="c3"
    >
      <select
        class="c4"
        id="myselect"
        name="myselect"
        required=""
      >
        <option
          value="option1"
        >
          option 1
        </option>
        <option
          selected=""
          value="option2"
        >
          option 2
        </option>
      </select>
      <svg
        height="14"
        viewBox="0 0 9 28"
        width="9"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.297 15c0 0.125-0.063 0.266-0.156 0.359l-7.281 7.281c-0.094 0.094-0.234 0.156-0.359 0.156s-0.266-0.063-0.359-0.156l-0.781-0.781c-0.094-0.094-0.156-0.219-0.156-0.359 0-0.125 0.063-0.266 0.156-0.359l6.141-6.141-6.141-6.141c-0.094-0.094-0.156-0.234-0.156-0.359s0.063-0.266 0.156-0.359l0.781-0.781c0.094-0.094 0.234-0.156 0.359-0.156s0.266 0.063 0.359 0.156l7.281 7.281c0.094 0.094 0.156 0.234 0.156 0.359z"
        />
      </svg>
    </div>
  </div>
</div>
`;
